src/main/java/org/italiangrid/storm/webdav/server/DefaultPathResolver.java

```diff
@@ -125,4 +135,57 @@ public Path getPath(String pathInContext) {
 
     return null;
   }
+
+  @Override
+  public boolean isStub(String pathInContext) {
+    String resolvedPath = resolvePath(pathInContext);
+    if (resolvedPath != null) {
+      File f = new File(resolvedPath);
+      if (f.isFile()) {
+        Stat stat = new Stat();
+        Libc.INSTANCE.stat(resolvedPath, stat);
+        long blockSize;
+        try {
+          blockSize = Files.getFileStore(f.toPath()).getBlockSize();
```

Comment:
there is no need to get the block size at each call. you can get it once during initialization, or you can even use the constant 512, which is what `stat` uses

Reply:
and in any case you could use the value from the Stat call that you have just done (but the constant is better)